src/components/Admin/Trainer/Input.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const Input = props => {
+	if (!props.hide)
+		return (
+			<div className="formGroup">
+				<label>
+					{props.label}:<br />
+					<input
+						type={props.type ? props.type : 'number'}
+						value={props.value}
+						onChange={e =>
+							props.setState('batchSizeFraction', e.target.value)
```

Comment:
This setState prop is being passed down from Trainer to Settings to Input - would it be neater to pass it in w/ Redux mapDispatchToProps?

Reply:
I agree, I think this component is close to being a good generic one. It's generic besides being specific to batch size fraction when changing.

We could pass this as a prop and let the parent care what it updates

e.g.

```
<input
  onChange={event => this.props.onChange(event.target.value)}
/>

<Input
  onChange={batchSizeFraction => this.setState({ batchSizeFraction })}
  label="Batch size"
/>
```

Though the parent might want the event for certain reasons so could just pass it directly